### DISCUSSION TITLE
Czy można uzyskiwać informacje z Allegro poprzez API w momencie, gdy aukcja na Allegro zostanie ręcznie wznowiona lub zakończona?

### DISCUSSION THREAD

**Discussion by DanielCyran** [allegro/allegro-api#8532] [Suggestions - rejected]
https://developer.allegro.pl/documentation/#operation/getOfferEvents

Posiadamy platformę do wysyłania produktów na Allegro poprzez API, i mam pytanie: czy można uzyskiwać informacje z Allegro poprzez API w momencie, gdy aukcja na Allegro zostanie ręcznie wznowiona lub zakończona?

Przede wszystkim chodzi o to, aby nie było konieczności regularnego odpytywania Allegro w celu sprawdzania, czy status danej aukcji został zmieniony. Czy istnieje możliwość skonfigurowania tak, aby Allegro wysyłało informacje do naszego systemu poprzez API, gdy status aukcji zostanie  recznie zmieniony po stronie Allegro?

**Comment by Lukasz-Zurek** [comment]
W dzienniku zdarzeń w ofertach [GET /sale/offer-events](https://developer.allegro.pl/documentation/#operation/getOfferEvents) zwracamy m.in. ogólne informacje o aktywowaniu czy zmianach w ofertach - nie rozróżniamy jednak w żaden sposób tych statusów. 

Natomiast w przypadku zakończenia oferty, w eventach zwracamy informacje dotyczące tego, kto zakończył ofertę, w polu **"publication.endedBy"**, np.

```
                "publication": {
                    "endedBy": "USER"
                },

```
co oznacza ręczne zakończenie oferty przez sprzedającego.

Nie jestem pewien jak bardzo szczegółowe statusy czy dodatkowe pola chciałbyś otrzymywać w eventach ofertowych. Oczywiście tu mogę Cię zachęcić do opisania propozycji i przesłania jej na [Github Discussions](https://github.com/allegro/allegro-api/discussions).

**Comment by DanielCyran** [comment]
@Lukasz-Zurek Ręczne wznowienie lub zakończenie oferty przez sprzedającego oznacza, że dokonuje zakończenia lub wznowienia oferty po stronie Allegro, a nie po stronie naszego oprogramowania.

Chcemy zabezpieczyć sprzedającego, aby w przypadku ręcznego wznowienia lub zakończenia aukcji po stronie Allegro, automatycznie dokonywały się odpowiednie zmiany w naszym systemie. Aby to osiągnąć, potrzebujemy uzyskiwać informacje od API Allegro, kiedy sprzedający wykonuje czynność wznowienia lub zakończenia aukcji ręcznie po stronie allegro.


Jeśli nasłuchujemy, czy otrzymamy informację, czy ktoś ręcznie zamknął lub wznowił aukcję, czy musimy specjalnie zapytać, aby uzyskać taką informację?

**Comment by Lukasz-Zurek** [comment]
@DanielCyran Ok, dziękuję. W pełni rozumiem przedstawioną przez Ciebie potrzebę. Statusów aktywacji czy zmiany w ofercie faktycznie nie można rozróżnić i to może stanowić pewien problem. Przenoszę Twój wątek na Github Discussions. Na pewno przekażę ten temat do odpowiedniego zespołu w celu dalszej analizy.

**Comment by MartaNowaczyk** [comment]
Zaktualizowaliśmy status sugestii jako "odrzucona", ponieważ nie planujemy wprowadzenia takiej zmiany.
